packages/mermaid/src/diagrams/xychart/parser/xychart.jison.spec.ts

```diff
@@ -419,30 +423,74 @@ describe('Testing xychart jison file', () => {
  line lineTitle1 [11, 45.5, 67, 23]
  bar barTitle2 [13, 42, 56.89]
     line lineTitle2 [45, 99, 012]`;
-    expect(parserFnConstructor(str)).not.toThrow();
-    expect(mockDB.setYAxisTitle).toHaveBeenCalledWith({ text: 'yaxisText', type: 'text' });
-    expect(mockDB.setYAxisRangeData).toHaveBeenCalledWith(10, 150);
-    expect(mockDB.setXAxisTitle).toHaveBeenCalledWith({ text: 'this is x axis', type: 'text' });
-    expect(mockDB.setXAxisBand).toHaveBeenCalledWith([
-      { text: 'category1', type: 'text' },
-      { text: 'category 2', type: 'text' },
-      { text: 'category3', type: 'text' },
-    ]);
-    expect(mockDB.setBarData).toHaveBeenCalledWith(
-      { text: 'barTitle1', type: 'text' },
-      [23, 45, 56.6]
-    );
-    expect(mockDB.setBarData).toHaveBeenCalledWith(
-      { text: 'barTitle2', type: 'text' },
-      [13, 42, 56.89]
-    );
-    expect(mockDB.setLineData).toHaveBeenCalledWith(
-      { text: 'lineTitle1', type: 'text' },
-      [11, 45.5, 67, 23]
-    );
-    expect(mockDB.setLineData).toHaveBeenCalledWith(
-      { text: 'lineTitle2', type: 'text' },
-      [45, 99, 12]
-    );
+      expect(parserFnConstructor(str)).not.toThrow();
+      expect(mockDB.setYAxisTitle).toHaveBeenCalledWith({ text: 'yaxisText', type: 'text' });
+      expect(mockDB.setYAxisRangeData).toHaveBeenCalledWith(10, 150);
+      expect(mockDB.setXAxisTitle).toHaveBeenCalledWith({ text: 'this is x axis', type: 'text' });
+      expect(mockDB.setXAxisBand).toHaveBeenCalledWith([
+        { text: 'category1', type: 'text' },
+        { text: 'category 2', type: 'text' },
+        { text: 'category3', type: 'text' },
+      ]);
+      expect(mockDB.setBarData).toHaveBeenCalledWith(
+        { text: 'barTitle1', type: 'text' },
+        [23, 45, 56.6]
+      );
+      expect(mockDB.setBarData).toHaveBeenCalledWith(
+        { text: 'barTitle2', type: 'text' },
+        [13, 42, 56.89]
+      );
+      expect(mockDB.setLineData).toHaveBeenCalledWith(
+        { text: 'lineTitle1', type: 'text' },
+        [11, 45.5, 67, 23]
+      );
+      expect(mockDB.setLineData).toHaveBeenCalledWith(
+        { text: 'lineTitle2', type: 'text' },
+        [45, 99, 12]
+      );
+    });
+  });
+
+  describe('multiple datasets', () => {
+    it('parse 2 datasets', () => {
+      const str =
+        'xychart-beta\nx-axis xAxisName\ny-axis yAxisName\nbar "barTitle1" [23, 45, 56.6]\nbar "barTitle2" [13, 42, 56.89]';
```

Comment:
Can you try if this works? It'll be more readable. 
```suggestion
        `xychart-beta
  x-axis xAxisName
  y-axis yAxisName
  bar "barTitle1" [23, 45, 56.6]
  bar "barTitle2" [13, 42, 56.89]`;
```

Reply:
Done